### ISSUE TITLE
SC2086 Not warning with default value assignment when empty string.

### ISSUE THREAD

**Issue by logand22** [koalaman/shellcheck#2403]
#### For bugs
- Rule Id: SC2086
- My shellcheck version: 0.8.0 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
MY_VAR=$1

if [ -z "$MY_VAR" ]; then
	MY_VAR=0
fi

printf 'MY_VAR is %s\n' $MY_VAR
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
```sh
printf 'MY_VAR is %s\n' $MY_VAR
                        ^-- SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
printf 'MY_VAR is %s\n' "$MY_VAR"
```

The removal of the `if..` block causes shellcheck to correctly produce the above error. When the `if..` block is present it doesn't warn you. 

Is this the intended behavior or a bug? 
Thanks
